Source/IO.cpp

```diff
@@ -1194,3 +1195,102 @@ PeleC::writeSmallPlotFile(
   TheFullPath += BaseName;
   amrex::VisMF::Write(plotMF, TheFullPath, how, true);
 }
+
+void
+PeleC::initLevelDataFromPlt(
+  const int lev, const std::string& dataPltFile, amrex::MultiFab& state)
+{
+  amrex::Print() << "Using data (rho, u, T, Y) from pltfile " << dataPltFile
+                 << std::endl;
+  pele::physics::pltfilemanager::PltFileManager pltData(dataPltFile);
+  const auto plt_vars = pltData.getVariableList();
+
+  // Read rho, u, temperature (required)
+  std::map<std::string, int> vars{
+    {"density", -1}, {"x_velocity", -1}, {"Temp", -1}};
+  for (auto& var : vars) {
+    var.second = find_position(plt_vars, var.first);
+    if (var.second == -1) {
+      amrex::Abort("Unable to find variable in plot file: " + var.first);
+    }
+  }
+  pltData.fillPatchFromPlt(lev, geom, vars["density"], URHO, 1, state);
```

Comment:
Another option would be to take the pressure from the plot file and use that to compute density. In an ideal world, these would be equivalent. But in a case where the various mass fraction modifications come into play, that might not be the case.

Reply:
Bruce and I talked about this. And for now this PR is fine.